src/KeyValueIndex.js

```diff
@@ -10,20 +10,24 @@ class KeyValueIndex {
   }
 
   updateIndex(oplog) {
-    oplog.values
-      .slice()
-      .reverse()
-      .reduce((handled, item) => {
-        if(!handled.includes(item.payload.key)) {
-          handled.push(item.payload.key)
-          if(item.payload.op === 'PUT') {
-            this._index[item.payload.key] = item.payload.value
-          } else if(item.payload.op === 'DEL') {
-            delete this._index[item.payload.key]
-          }
-        }
-        return handled
-      }, [])
+    const values = oplog.values;
+
+    const handled = {};
```

Comment:
great catch, thought this was already the case. this probably needs to be changed with the docstore as well (dont feel obligated).

Reply:
I was already planning to update the docstore as well if this got accepted.